### DISCUSSION TITLE
REST API - POST request body

### DISCUSSION THREAD

**Discussion by jeremiah-blessing** [Budibase/budibase#3802] [Help]
I am sending Budibase form data to an external POST request API.  The schema looks like below

```
{
description : String,
countries : Array,
type : String,
subType : String
}
```

In that Budibase form, I have a  **text field** called `countries`. In that field, the user can type by commas like `US,UK,CH`. While submitting I need to split this string and **send it as an array field** instead of a string(Screenshot below).

Thanks!

Screenshots :

![image](https://user-images.githubusercontent.com/47788987/146351130-c7978391-358d-4fcf-a79a-44ddff3b1459.png)

**Comment by mjashanks** [comment]
You will need to do `split` when you are passing parameters in from the design section:

![image](https://user-images.githubusercontent.com/3524181/146359148-ed66afb1-71ce-4068-9875-49648c08246f.png)

then...

![image](https://user-images.githubusercontent.com/3524181/146359594-c5578a3d-3c60-4e76-b102-ba74388cdebb.png)

**Comment by jeremiah-blessing** [comment]
![image](https://user-images.githubusercontent.com/47788987/146364214-55d06230-9c1d-4d1f-8f7f-2367de431149.png)

I did this and then in the request body, the array is automatically turned into a string in request body.

Eg: 
| Value       | State                               |
|-------------|-------------------------------------|
| US,UK       | In Budibase Input Component         |
| ["US","UK"] | After splitting by using Javascript |
| "US,UK"     | REST API Parser                     |

**Reply by mjashanks** [reply]
ah. I see.

Try passing the stringified array ... i.e. from the `Design` section....

```javascript
return JSON.stringify($("Form.Fields.countries").split(","));
```

**Reply by mjashanks** [reply]
and your REST body is

```
{
 ....
 "countries": {{ countries }},
...
}
```

i.e. no quotes around the parameters

**Reply by jeremiah-blessing** [reply]
I tried this but didn't work though. I will try other methods if I am able to fix this. Thanks : )

**Comment by oswaldB** [comment]
Hello,

I face exaclty the same problem and it is not working...

any idea?

Tahnk you

**Reply by shogunpurple** [reply]
Can you provide some more details please @oswaldB ?

**Reply by oswaldB** [reply]
Hello I succeeded by doing so:

let a = $("new_dossier_form.Fields.test-toto");
b = a.map(x=>parseInt(x))
return "["+b.join(',')+"]"

**Comment by ShinJustinHolly3317** [comment]
hey try using 

```
{
    countries: [ {{ countries }} ]
}
```

and in your form submit actions you just keep the same

```
$("Form.Fields.countries").split(",")
```

**Reply by ShinJustinHolly3317** [reply]
I think it's a bug in budibase, no idea why it changes into string

**Comment by tugay-radity** [comment]
Using a simple helper function did the trick. Directly returning an array in bindings doesn't work. 

```JavaScript

function arrayToString(arr) {
  return `[${arr.map(item => `"${item}"`).join(", ")}]`;
}

return arrayToString($("Create User Form.Fields.roles"))
```
